File: SuncoLab.Model/Database/Album.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace SuncoLab.Model
{
    public class Album : BaseEntity
    {
        public string Name { get; set; }
        public string? Description { get; set; }
        public bool Show { get; set; }

        [ForeignKey("CoverImage")]
        public Guid? CoverImageId { get; set; }
        public Image CoverImage { get; set; }
    }
}

File: SuncoLab.Frontend/src/app/models/album.ts
import { Image } from "./image";

export class Album {
    id: string;
    name: string;
    description: string | undefined;
    show: boolean;
    coverImage?: Image;

    constructor(id: string, name: string, description: string | undefined, show: boolean, coverImage: Image) {
    this.id = id;
    this.name = name;
    this.description = description;
    this.show = show;
    this.coverImage = coverImage;
  }
}

export class AlbumFilter {
  visibleOnly: boolean = false;
  name: string | undefined;

  constructor(visibleOnly?: boolean, name?: string) {
    this.visibleOnly = visibleOnly ?? false;
    this.name = name;
  }
}
